_applications/equoid.adoc

```diff
@@ -0,0 +1,206 @@
+
+The services of Equoid are: 
+
+- A data publisher
+- A message broker
+- A data handler
+- A caching layer 
+- A web UI
+- A REST API for adding items to the set of possibilities the data publisher chooses from 
+
+The following diagram shows the overall architecture of Equoid.
+
+pass:[<img src="/assets/equoid/Equoid-Architecture.png" alt="Alt text" class="img-responsive arch" width="800px">]
+
+Equoid utilizes a caching layer for storing the top-k values. The data publisher transmits records to Artemis, the data handler retrieves mesages from the Artemis broker, parses or converts them per user specification, then adds the information to the extant top-k object. It's worth noting that multiple instances of the data publisher and data handler can co-exist, which would be akin to a production system. 
+
+== Software Components
+
+We chose to use Artemis for the message broker, with the expectation it would be scalable in terms of number of streams as well as volume of traffic. In the same vein, we utilized Spark for our microservice requiring scalability, namely the data handler. Since we wished to take advantage of extant Spark libraries, and the CMSketch() object is only (as of this writing) available in Java and Scala, we chose to implement the handler in Scala. In the interest of uniformity the sample data publisher is also written in Scala. Furthermore, the AMQP Spark connector used by the data handler is one developed by Paolo Patierno at Red Hat, with more information https://radanalytics.io/examples/amqpstreaming[this tutorial] and associated https://github.com/redhat-iot/amqp-spark-demo[upstream project].
+
+pass:[<img src="/assets/equoid/Equoid-Software-Components.png" alt="Alt text" class="img-responsive arch" width="800px">]
+ 
+[[installation]]
+== Installation
+
+As mentioned previously, Equoid consists of a number of microservices which require deployment to an OpenShift instance. In order to get you started as swiftly as possible, in this section a walkthrough of installing and configuring the Equoid services is given. These should be followed in the order presented. These steps follow the same flow as the equoid-openshift repository's https://github.com/EldritchJS/equoid-openshift/blob/master/start-full.sh[start-full.sh] script. 
+
+[[prerequisites]]
+=== Prerequisites
+
+You will need an OpenShift instance in place which you are able to access and create new projects on. This could be local using minishift https://docs.okd.io/latest/minishift/getting-started/quickstart.html[(tutorial here)] or the oc tooling https://github.com/openshift/origin/releases[(download found here)], or on an admistered cluster to which you have access. Be certain to log in to the OpenShift cluster, then you can created your project by running the following command:
```

Comment:
My English sucks, but `then you can created` -?> `then you can create`

Reply:
Clearly in this case it's my English that sucks. Nice catch, thanks.